clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -545,6 +545,20 @@ class MarkDeviceFunction : public RecursiveASTVisitor<MarkDeviceFunction> {
 
       if (auto *A = FD->getAttr<SYCLSimdAttr>())
         Attrs.insert(A);
+
+      // Allow the kernel attribute "stall_enable" only on lambda functions
+      // and function objects that are called directly from a kernel
+      // (i.e. the one passed to the single_task or parallel_for functions).
+      // For all other cases, emit a warning and ignore.
+      if (auto *A = FD->getAttr<SYCLIntelStallEnableAttr>()) {
```

Comment:
Did the tests pass with this change? I thought ParentFD here is the caller.

Reply:
I run check-all and the tests pass with this change.

Reply:
Ok. I spend some time trying to understand how this works. The parent is SYCLKernel when FD is operator() or lambda function. So this is correct.